kyo-core/shared/src/test/scala/kyo/ChannelTest.scala

```diff
@@ -347,34 +360,34 @@ class ChannelTest extends Test:
                 s <- c.size
             yield assert(r == Seq(1, 2) && s == 2)
         }
-        "should consider pending puts" in pendingUntilFixed {
+        "should consider pending puts" in run {
             import AllowUnsafe.embrace.danger
             IO.Unsafe.evalOrThrow {
                 for
                     c         <- Channel.init[Int](2)
                     _         <- c.putFiber(1)
                     _         <- c.putFiber(2)
                     _         <- c.putFiber(3)
+                    _         <- c.putFiber(4)
                     result    <- c.drainUpTo(3)
                     finalSize <- c.size
-                yield assert(result == Chunk(1, 2, 3) && finalSize == 0)
-            }
-            ()
-        }
-        "should consider pending puts - zero capacity" in pendingUntilFixed {
-            import AllowUnsafe.embrace.danger
-            IO.Unsafe.evalOrThrow {
-                for
-                    c         <- Channel.init[Int](0)
-                    _         <- c.putFiber(1)
-                    _         <- c.putFiber(2)
-                    _         <- c.putFiber(3)
-                    result    <- c.drainUpTo(3)
-                    finalSize <- c.size
-                yield assert(result == Chunk(1, 2, 3) && finalSize == 0)
+                yield assert(result == Chunk(1, 2, 3) && finalSize == 1)
             }
-            ()
         }
+        // "should consider pending puts - zero capacity" in run {
```

Comment:
can you keep this test pending and open an issue to handle channels with zero capacity separately? It'd be possible to provide better performance with a dedicated impl as well